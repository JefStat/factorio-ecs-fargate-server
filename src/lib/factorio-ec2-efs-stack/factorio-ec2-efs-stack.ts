import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { createEc2 } from "./supportingCode/ec2";

export class FactorioEc2EfsStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    createEc2(this);
  }
}