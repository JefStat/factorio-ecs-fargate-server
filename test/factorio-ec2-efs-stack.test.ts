import './test-env';
import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from 'aws-cdk-lib';
import { FactorioEc2EfsStack } from '../src/lib/factorio-ec2-efs-stack/factorio-ec2-efs-stack';

const env = {
  region: 'us-east-1',
  account: '111222333444',
};

describe('FactorioEc2EfsStack', () => {
  test('Synths Stack', async () => {
    // GIVEN
    const app = new cdk.App();
    const stack = new FactorioEc2EfsStack(app, 'MyTestStack', {env});
    // WHEN
    const synth = SynthUtils.toCloudFormation(stack);
    // THEN
    expect(synth).toMatchSnapshot();
  });
});
