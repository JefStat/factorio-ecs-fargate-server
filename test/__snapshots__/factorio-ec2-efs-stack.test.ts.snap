// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactorioEc2EfsStack Synths Stack 1`] = `
{
  "Parameters": {
    "SsmParameterValueawsserviceamiamazonlinuxlatestal2023amikernel61x8664C96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-6.1-x86_64",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
    },
  },
  "Resources": {
    "KeyPair1622897B": {
      "Properties": {
        "KeyFormat": "pem",
        "KeyName": "MyTestStackKeyPairEDAD48BF",
        "KeyType": "rsa",
      },
      "Type": "AWS::EC2::KeyPair",
    },
    "efsaccessec204B16FDD": {
      "DependsOn": [
        "efsaccessec2InstanceRole01572541",
      ],
      "Properties": {
        "AvailabilityZone": "dummy1a",
        "IamInstanceProfile": {
          "Ref": "efsaccessec2InstanceProfile23C615DF",
        },
        "ImageId": {
          "Ref": "SsmParameterValueawsserviceamiamazonlinuxlatestal2023amikernel61x8664C96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "InstanceType": "t2.micro",
        "KeyName": {
          "Ref": "KeyPair1622897B",
        },
        "NetworkInterfaces": [
          {
            "AssociatePublicIpAddress": true,
            "DeviceIndex": "0",
            "GroupSet": [
              {
                "Fn::GetAtt": [
                  "factorioec2efsaccesssecuritygroup2B5EBD25",
                  "GroupId",
                ],
              },
            ],
            "SubnetId": "s-12345",
          },
        ],
        "Tags": [
          {
            "Key": "Name",
            "Value": "undefined-factorio-efs-access",
          },
        ],
        "UserData": {
          "Fn::Base64": "#!/bin/bash",
        },
      },
      "Type": "AWS::EC2::Instance",
    },
    "efsaccessec2InstanceProfile23C615DF": {
      "Properties": {
        "Roles": [
          {
            "Ref": "efsaccessec2InstanceRole01572541",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "efsaccessec2InstanceRole01572541": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "undefined-factorio-efs-access",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "factorioec2efsaccesssecuritygroup2B5EBD25": {
      "Properties": {
        "GroupDescription": "MyTestStack/factorio-ec2-efs-access-security-group",
        "GroupName": "undefined-factorio-ec2-efs-access-security-group",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "SecurityGroupIngress": [
          {
            "CidrIp": "127.0.0.1/32",
            "Description": "Allow SSH access from local",
            "FromPort": 22,
            "IpProtocol": "tcp",
            "ToPort": 22,
          },
        ],
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
  },
}
`;
